RecoEgamma/PhotonIdentification/plugins/PhotonMVAEstimatorRunIIFall17.cc

```diff
@@ -0,0 +1,312 @@
+  
+  allMVAVars.varRawE         = superCluster->rawEnergy(); 
+  allMVAVars.varR9           = phoRecoPtr->r9() ;
+  allMVAVars.varSieie        = full5x5_sigmaIetaIeta; 
+  allMVAVars.varSCEtaWidth   = superCluster->etaWidth(); 
+  allMVAVars.varSCPhiWidth   = superCluster->phiWidth(); 
+  allMVAVars.varSieip        = full5x5_sigmaIetaIphi;
+  allMVAVars.varE2x2overE5x5 = e2x2/e5x5;
+  allMVAVars.varPhoIsoRaw    = (*phoPhotonIsolationMap)[phoRecoPtr];  
+  allMVAVars.varChIsoRaw     = (*phoChargedIsolationMap)[phoRecoPtr];
+  allMVAVars.varWorstChRaw   = (*phoWorstChargedIsolationMap)[phoRecoPtr];
+  allMVAVars.varSCEta        = superCluster->eta(); 
+  allMVAVars.varRho          = *rho; 
+  allMVAVars.varESEffSigmaRR = effSigmaRR;
+  allMVAVars.varESEnOverRawE = superCluster->preshowerEnergy() / superCluster->rawEnergy();
+  
+  constrainMVAVariables(allMVAVars);
+  //
+  // Important: the order of variables in the "vars" vector has to be EXACTLY
+  // the same as in the .xml file defining the MVA.
+  //
+  std::vector<float> vars;
+  if( isEndcapCategory( findCategory( particle ) ) ) {
+    vars = packMVAVariables(
+                                       allMVAVars.varRawE,
+                                       allMVAVars.varR9,
+                                       allMVAVars.varSieie,
+                                       allMVAVars.varSCEtaWidth,
+                                       allMVAVars.varSCPhiWidth,
+                                       allMVAVars.varSieip,
+                                       allMVAVars.varE2x2overE5x5,
+                                       allMVAVars.varPhoIsoRaw,
+				       allMVAVars.varChIsoRaw,
+				       allMVAVars.varWorstChRaw,
+                                       allMVAVars.varSCEta,
+				       allMVAVars.varRho,
+				       allMVAVars.varESEffSigmaRR,
+				       allMVAVars.varESEnOverRawE
+				       ) 
+                      ; 
+  } else {
+    vars = packMVAVariables(
+                                       allMVAVars.varRawE,
+                                       allMVAVars.varR9,
+                                       allMVAVars.varSieie,
+                                       allMVAVars.varSCEtaWidth,
+                                       allMVAVars.varSCPhiWidth,
+                                       allMVAVars.varSieip,
+                                       allMVAVars.varE2x2overE5x5,
+                                       allMVAVars.varPhoIsoRaw,
+				       allMVAVars.varChIsoRaw,
+				       allMVAVars.varWorstChRaw,
+                                       allMVAVars.varSCEta,
+				       allMVAVars.varRho
+				       ) 
+                      ;
+  }
+  return vars;
+}
+
+void PhotonMVAEstimatorRunIIFall17::constrainMVAVariables(AllVariables&)const {
+  
+  // Check that variables do not have crazy values
+  
+  // This function is currently empty as this specific MVA was not
+  // developed with restricting variables to specific physical ranges.
+  return;
```

Comment:
is this return needed?

Reply:
Changes implemented here:
https://github.com/lsoffi/cmssw/blob/CMSSW_9_4_0_pre3_TnP/RecoEgamma/PhotonIdentification/plugins/PhotonMVAEstimatorRunIIFall17.cc